### PR TITLE
Extension: Prepare stream for all domains in a hls playlist

### LINKED ISSUE #905: [Bug]: Extension does not work for Closeload
### What happened?

The Closeload embed does not work with the extension. This is because there are different domains in the hls playlist files. Only the main domain is prepared through the extension. As you can see in the screenshot, a domain rule is set for hls.closeload, but not for  srv9.servercdn1035

### What browsers are you seeing the problem on?

_No response_

### Steps to reproduce?

1. Use extension v1.1.0
2. Choose a movie, and Ridomovies

### Other relevant information

image
image
image
image

### PR DESCRIPTION
**Pull request by JorrinKievit**
This pull request resolves #905 and VidSrcTo

There was also a bug in the hlsLevelQuality function, sometimes levels don't exist and it throws an error which causes the code after not to run.

 - [x] I have read and agreed to the [code of conduct](https://github.com/movie-web/movie-web/blob/dev/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/movie-web/movie-web/blob/dev/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me and is an [approved issue](https://github.com/movie-web/movie-web/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved). For reference, please take a look at our [GitHub projects](https://github.com/movie-web/movie-web/projects).
 - [x] I have tested all of my changes.

### CODE REVIEW

**Comment by sudo-flix** [comment]
looks fire

File: src/components/player/display/base.ts [hunk 1]
```diff
@@ -144,6 +149,24 @@ export function makeVideoElementDisplayInterface(): DisplayInterface {
           if (!hls) return;
           reportLevels();
           setupQualityForHls();
+
+          if (isExtensionActiveCached()) {
+            hls.on(Hls.Events.LEVEL_LOADED, async (_, data) => {
+              const chunkUrlsDomains = data.details.fragments.map(
+                (v) => new URL(v.url).hostname,
```
Comment (binaryoverload) [inline src/components/player/display/base.ts:156]: Is there any chance of a URL being undefined/invalid format on a fragment - worth handling?

**Review by binaryoverload — CHANGES_REQUESTED** [review]

File: src/components/player/display/base.ts (hunk 1 above)
Comment (JorrinKievit) [reply src/components/player/display/base.ts:156]: Good point, not sure if hls.js handles that. Will check when I get the time (or if someone else wants to 😉)

Comment (qtchaos) [reply src/components/player/display/base.ts:156]: Would something like this be enough? If the URL is null we just use an empty string which will make the request 403 (black screen on player)

```ts
const chunkUrlsDomains = data.details.fragments.map(
  (v) => new URL(v.url).hostname ?? "",
);
```

Comment (JorrinKievit) [reply src/components/player/display/base.ts:156]: Issue is more that new URL() can throw an error if the input is invalid

**Review by qtchaos — APPROVED** [review]

**Pull request merged**